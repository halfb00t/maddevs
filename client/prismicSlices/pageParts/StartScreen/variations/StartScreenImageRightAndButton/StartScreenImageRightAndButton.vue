<template>
  <section
    id="transparent-header-area"
    data-testid="start-screen-slice-with-image"
    class="start-screen-slice start-screen-slice-with-image"
    :style="{ background: sliceGradient }"
  >
    <img
      :src="image && image.url"
      :alt="image && image.alt || 'Image'"
      width="1680"
      height="969"
      :style="{
        opacity: imageOpacity,
        backgroundColor: sliceBackground,
      }"
      data-testid="start-screen-slice__image"
      class="start-screen-slice__image"
    >
    <!-- sectionTextOpacity - value from changeOpacityOnScrollMixin mixin -->
    <div class="container">
      <div
        ref="sectionText"
        :style="{ opacity: sectionTextOpacity }"
        class="start-screen-slice__flex-wrapper"
      >
        <div
          class="start-screen-slice__content"
        >
          <div class="start-screen-slice__titles-wrapper">
            <PrismicRichText
              v-if="$prismic.asText(title)"
              :field="title"
              data-testid="start-screen-slice__title"
              class="start-screen-slice__title"
              :html-serializer="serializer"
            />
          </div>
          <PrismicRichText
            v-if="$prismic.asText(subtitle)"
            data-testid="start-screen-slice__subtitle"
            class="start-screen-slice__subtitle"
            :field="subtitle"
            :html-serializer="serializer"
          />
          <div class="start-screen-slice__btn-box">
            <UIButton
              v-if="modal === 'download-pdf'"
              class="start-screen-slice__btn"
              @click="showModalEbook"
            >
              {{ btnText }}
              <img
                width="20"
                height="20"
                src="@/assets/img/common/arrow-up-right.svg"
                alt="Arrow"
              >
            </UIButton>
          </div>
        </div>
        <div
          class="start-screen-slice__img-box"
        >
          <img
            :src="rightImage && rightImage.url"
            :alt="rightImage && rightImage.alt || 'Image'"
            width="259.69"
            height="380.25"
            class="start-screen-slice__img-box-image"
            data-testid="right-image"
          >
        </div>
      </div>
    </div>
    <ModalEbook
      id="sent-ebook"
      ref="modalEbook"
      :ebook-title="$prismic.asText(title)"
      :ebook-path="ebookPath"
      :book-name="bookName"
      :book-image="ebookImage"
      :send-pulse-template-id="sendPulseTemplateId"
    />
  </section>
</template>

<script>
import changeSectionTextOpacityMixin from '@/mixins/changeSectionTextOpacityMixin'
import setSliceBackground from '@/helpers/setSliceBackground'
import UIButton from '@/components/shared/UIButton'
import ModalEbook from '@/components/core/modals/ModalEbook'
import convertTagsToText from '@/helpers/convertTagsToText'
import linkResolver from '@/plugins/link-resolver'

export default {
  name: 'StartScreen',
  components: {
    UIButton,
    ModalEbook,
  },

  mixins: [changeSectionTextOpacityMixin('sectionText')],
  props: {
    slice: {
      type: Object,
      required: true,
      default() {
        return {}
      },
    },
  },

  data() {
    let sliceColor = ''
    if (this.slice?.primary?.gradientColor === 'white') sliceColor = 'linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #ffffff)'
    if (this.slice?.primary?.gradientColor === 'grey') sliceColor = 'linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #f5f7f9)'
    if (this.slice?.primary?.gradientColor === 'black') sliceColor = 'linear-gradient(180deg, rgba(17, 18, 19, 0) 0, #111213 100%)'

    return {
      imageOpacity: this.slice?.primary?.imageOpacity,
      image: this.slice?.primary?.image,
      title: this.slice?.primary?.title,
      subtitle: this.slice?.primary?.subtitle,
      sliceBackground: setSliceBackground(this.slice?.primary?.background),
      sliceGradient: sliceColor,
      rightImage: this.slice?.primary?.rightImage,
      btnText: this.slice?.primary?.btnText,
      btnLink: this.slice?.primary?.btnLink,
      modal: this.slice?.primary?.modal,
      ebookPath: this.slice?.primary?.bookPath,
      bookName: this.slice?.primary?.bookName,
      ebookImage: this.slice?.primary?.ebookImage,
      sendPulseTemplateId: this.slice?.primary?.sendPulseTemplateId,
    }
  },

  computed: {
    serializer() {
      const types = ['start_screen_slice']
      if (types.includes(this.slice?.slice_type)) return this.htmlSerializer
      return null
    },
  },

  methods: {
    showModalEbook() {
      if (!this.$refs?.modalEbook?.show) return
      this.$refs.modalEbook.show()
    },

    htmlSerializer(type, element, content, children) {
      const { Elements } = this.$prismic.dom.RichText
      const { Link } = this.$prismic.dom
      let text = children.join('')

      if (type === Elements.preformatted) {
        // the second parameter of function excludes tags
        text = convertTagsToText(text, ['br'])
      } else {
        text = text.replace(/`(.*?)`/g, (_, inlineCode) => {
          // the second parameter of function excludes tags
          const formattedCode = convertTagsToText(inlineCode, ['strong', 'em', 'a'])
          return `<code class="inline-code">${formattedCode}</code>`
        })
      }

      switch (type) {
        case Elements.heading1:
          return `<h1>${text}</h1>`
        case Elements.heading2:
          return `<h2>${text}</h2>`
        case Elements.heading3:
          return `<h3>${text}</h3>`
        case Elements.heading4:
          return `<h4>${text}</h4>`
        case Elements.heading5:
          return `<h5>${text}</h5>`
        case Elements.heading6:
          return `<h6>${text}</h6>`
        case Elements.paragraph:
          return `<p>${text}</p>`
        case Elements.preformatted:
          return `<pre>${text}</pre>`
        case Elements.strong:
          return `<strong>${text}</strong>`
        case Elements.em:
          return `<em>${text}</em>`
        case Elements.listItem:
          return `<li>${text}</li>`
        case Elements.oListItem:
          return `<li>${text}</li>`
        case Elements.list:
          return `<ul>${text}</ul>`
        case Elements.oList:
          return `<ol>${text}</ol>`
        case Elements.image:
          // eslint-disable-next-line
          const linkUrl = element.linkTo ? Link.url(element.linkTo, linkResolver) : null
          // eslint-disable-next-line
          const linkTarget = element.linkTo && element.linkTo.target ? `target="${element.linkTo.target}" rel="noopener"` : ''
          // eslint-disable-next-line
          const wrapperClassList = [element.label || '', 'block-img']
          // eslint-disable-next-line
          const img = `<img src="${element.url}" alt="${element.alt || 'Image'}" copyright="${element.copyright || ''}">`
          return (`
            <p class="${wrapperClassList.join(' ')}">
              ${linkUrl ? `<a ${linkTarget} href="${linkUrl}">${img}</a>` : img}
            </p>
          `)
        case Elements.embed:
          return (`
            <div data-oembed="${element.oembed.embed_url}"
              data-oembed-type="${element.oembed.type}"
              data-oembed-provider="${element.oembed.provider_name}"
            >
              ${element.oembed.html}
            </div>
          `)
        case Elements.hyperlink:
          // eslint-disable-next-line
          const target = element.data.target ? `target="${element.data.target}" rel="noopener"` : ''
          // eslint-disable-next-line
          const url = Link.url(element.data, linkResolver)
          return `<a ${target} href="${url}">${text}</a>`
        case Elements.label:
          // eslint-disable-next-line
          const label = element.data.label ? ` class="${element.data.label}"` : ''
          return `<span ${label}>${text}</span>`
        case Elements.span:
          return content ? content.replace(/\n/g, '<br />') : ''
        default:
          return null
      }
    },
  },
}
</script>

<style lang="scss" scoped>
.start-screen-slice {
  background: linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #111213);
  overflow: hidden;
  padding-top: 268px;
  height: auto;
  @media screen and (max-width: 768px) {
    padding-top: 147px;
  }

  &__image {
    display: block;
    z-index: -1;
    position: absolute;
    top: -2px;
    left: -2px;
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    object-fit: cover;
    background-color: $bgcolor--black;
  }

  &__title {
    color: $text-color--white-primary;
    margin-bottom: 70px;
    max-width: 650px;
    @media screen and (max-width: 580px) {
      margin-bottom: 32px;
    }

    ::v-deep h1,
    ::v-deep h2,
    ::v-deep h3,
    ::v-deep h4,
    ::v-deep h5,
    ::v-deep h6 {
      @include font('Inter', 64px, 700);
      line-height: 72px;
      letter-spacing: -2px;
      @media screen and (max-width: 580px) {
        @include font('Inter', 42px, 600);
        line-height: 48px;
      }
    }
  }

  &__subtitle {
    @include font('Inter', 32px, 600);
    line-height: 44px;
    letter-spacing: -1px;
    color: $text-color--white-primary;
    max-width: 704px;
    margin-bottom: 50px;

    @media screen and (max-width: 1024px) {
      font-size: 32px;
      line-height: 44px;
    }

    @media screen and (max-width: 768px) {
      font-size: 24px;
      line-height: 35px;
    }

    @media screen and (max-width: 580px) {
      font-size: 24px;
      line-height: 35px;
    }
  }

  ::v-deep br {
    @media only screen and (max-width: 520px) {
      display: none;
    }
  }

  &__flex-wrapper {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    @media screen and (max-width: 768px) {
      flex-direction: column-reverse;
    }
  }

  &__content {
    margin-right: 30px;
    display: flex;
    flex-direction: column;
    @media screen and (max-width: 768px) {
      width: 100%;
    }
  }

  &__img-box {
    width: 25%;
    position: relative;
    top: -60px;
    @media screen and (min-width: 1025px) {
      margin-right: 75px;
    }
    @media screen and (max-width: 1024px) {
      width: 45%;
    }
    @media screen and (max-width: 768px) {
      width: 211px;
      margin-bottom: 83px;
    }
    @media screen and (max-width: 580px) {
      margin-bottom: 47px;
    }
  }

  &__img-box-image {
    width: 100%;
    height: auto;
    display: block;
  }

  &__btn-box {
    width: 100%;
    display: flex;

    a,
    button {
      @include font('Inter', 16px, 600);
      letter-spacing: -0.4px;
      line-height: 20px;
      color: $text-color--white-primary;
      margin-right: auto;
    }

    button {
      display: flex;
      align-items: center;
      padding: 14px 132.5px;
      margin-right: 32px;
      @media screen and (max-width: 580px) {
        padding: 14px 24px;
      }

      @media screen and (max-width: 880px) {
        padding: 14px 100px;
      }

      img {
        margin-left: 7px;
      }
    }

    a {
      text-decoration-line: underline;
      text-underline-offset: 2px;
    }
  }
}
</style>
