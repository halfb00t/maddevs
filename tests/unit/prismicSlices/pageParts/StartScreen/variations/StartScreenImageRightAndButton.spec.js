import { render, screen } from '@testing-library/vue'
import { RouterLinkStub, shallowMount } from '@vue/test-utils'
import StartScreenImageRightAndButton
  from '@/prismicSlices/pageParts/StartScreen/variations/StartScreenImageRightAndButton/StartScreenImageRightAndButton'

const toOneLine = text => text
  .replace(/ +(?= )/g, '')
  .replace(/\n/g, '')
  .replace(/>(.*?)</g, '><')
  .trim()

const backgrounds = {
  white: '#ffffff',
  black: '#111213',
  grey: '#f5f7f9',
}

jest.mock('~/helpers/generatorUid')

const gradients = {
  black: 'linear-gradient(180deg, rgba(17, 18, 19, 0) 0, #111213 100%)',
  white: 'linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #ffffff)',
  grey: 'linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #f5f7f9)',
}

const startScreenEbookData = {
  slice: {
    primary: {
      imageOpacity: 0.9,
      image: {
        url: 'img.jpg',
        alt: 'img',
      },
      title: [{
        text: 'Test title',
      }],
      subtitle: [{
        text: 'Test subtitle',
      }],
      background: backgrounds.black,
      gradientColor: gradients.black,
      rightImage: {
        url: 'img.jpg',
        alt: 'img',
      },
      btnText: 'Test text',
      btnLink: '',
      modal: 'download-pdf',
      ebookPath: 'pdf/test-ebook.pdf',
      ebookName: 'test-ebook',
      ebookImage: {
        url: 'img.jpg',
        alt: 'Image',
      },
    },
    variation: 'startScreenImageRightAndButton',
    slice_type: 'start_screen_slice',
  },
}

const mocks = {
  $prismic: {
    asText: text => text[0].text,
  },
}

const stubs = ['UIButton', 'PrismicRichText']

let mockShow = null

describe('StartScreenImageRightAndButton component', () => {
  let wrapper = null

  beforeEach(() => {
    mockShow = jest.fn()

    wrapper = shallowMount(StartScreenImageRightAndButton, {
      propsData: startScreenEbookData,
      mocks,
      stubs,
    })
  })

  afterEach(() => {
    mockShow = null
    wrapper = null
  })

  it('should correctly render startScreenImageRightAndButton', () => {
    const { container } = render(StartScreenImageRightAndButton, {
      propsData: startScreenEbookData,
      mocks,
      stubs,
    })

    expect(screen.queryByTestId('start-screen-slice__title')).not.toBeNull()
    expect(screen.queryByTestId('start-screen-slice__subtitle')).not.toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('should return empty object from slice props', () => {
    expect(wrapper.vm.$options.props.slice.default.call()).toEqual({})
  })

  describe('sliceGradient computed method', () => {
    it(`should return ${backgrounds.black} (black gradient)`, () => {
      startScreenEbookData.slice.primary.gradientColor = 'black'
      wrapper = shallowMount(StartScreenImageRightAndButton, {
        propsData: startScreenEbookData,
        mocks,
        stubs,
      })
      expect(wrapper.vm.sliceGradient).toBe(gradients.black)
    })

    it(`should return ${backgrounds.white} (white gradient)`, () => {
      startScreenEbookData.slice.primary.gradientColor = 'white'
      wrapper = shallowMount(StartScreenImageRightAndButton, {
        propsData: startScreenEbookData,
        mocks,
        stubs,
      })
      expect(wrapper.vm.sliceGradient).toBe(gradients.white)
    })

    it(`should return ${backgrounds.grey} (grey gradient)`, () => {
      startScreenEbookData.slice.primary.gradientColor = 'grey'
      wrapper = shallowMount(StartScreenImageRightAndButton, {
        propsData: startScreenEbookData,
        mocks,
        stubs,
      })
      expect(wrapper.vm.sliceGradient).toBe(gradients.grey)
    })
  })

  it('should return null from serializer computed method', () => {
    wrapper.vm.slice.slice_type = 'test'
    expect(wrapper.vm.serializer).toBeNull()
  })

  it('if call method "showModalEbook" and ref modalEbook is undefined > "showModalEbook" method in ref modalEbook not call', () => {
    wrapper.vm.showModalEbook()
    expect(mockShow).not.toHaveBeenCalled()
  })

  it('should open modal window send ebook', () => {
    const wrapperTestModal = shallowMount(StartScreenImageRightAndButton, {
      propsData: startScreenEbookData,
      mocks,
      stubs: {
        NuxtLink: RouterLinkStub,
        PrismicRichText: {
          render(h) { return h('div') },
        },
        ModalEbook: {
          render(h) { return h('div') },
          methods: {
            show: mockShow,
          },
        },
      },
    })

    wrapperTestModal.vm.showModalEbook()
    expect(mockShow).toHaveBeenCalledTimes(1)
  })
})

describe('htmlSerializer', () => {
  let wrapper = null

  beforeEach(() => {
    wrapper = shallowMount(StartScreenImageRightAndButton, {
      propsData: startScreenEbookData,
      stubs,
      mocks: {
        $prismic: {
          asText: text => text[0].text,
          dom: {
            Link: {
              url: () => '',
            },
            RichText: {
              Elements: {
                em: 'em',
                embed: 'embed',
                heading1: 'heading1',
                heading2: 'heading2',
                heading3: 'heading3',
                heading4: 'heading4',
                heading5: 'heading5',
                heading6: 'heading6',
                hyperlink: 'hyperlink',
                image: 'image',
                label: 'label',
                list: 'group-list-item',
                listItem: 'list-item',
                oList: 'group-o-list-item',
                oListItem: 'o-list-item',
                paragraph: 'paragraph',
                preformatted: 'preformatted',
                span: 'span',
                strong: 'strong',
              },
            },
          },
        },
      },
    })
  })

  afterEach(() => {
    wrapper = null
  })

  it('if type not heading will return null', () => {
    const result = wrapper.vm.htmlSerializer('div', null, null, ['StartScreenImage title'])
    expect(result).toBeNull()
  })

  it('if type heading1 will return html with h1 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading1', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h1>StartScreenImage title</h1>')
  })

  it('if type heading2 will return html with h2 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading2', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h2>StartScreenImage title</h2>')
  })

  it('if type heading3 will return html with h3 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading3', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h3>StartScreenImage title</h3>')
  })

  it('if type heading4 will return html with h4 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading4', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h4>StartScreenImage title</h4>')
  })

  it('if type heading5 will return html with h5 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading5', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h5>StartScreenImage title</h5>')
  })

  it('if type heading6 will return html with h6 tag', () => {
    const result = wrapper.vm.htmlSerializer('heading6', null, null, ['StartScreenImage title'])
    expect(result).toBe('<h6>StartScreenImage title</h6>')
  })

  it('if type heading6 will return html with pre code text', () => {
    const result = wrapper.vm.htmlSerializer('heading6', null, null, ['`<em>ololo</em>` Custom page card title'])
    expect(result).toBe('<h6><code class="inline-code"><em>ololo</em></code> Custom page card title</h6>')
  })

  it('if type paragraph will return <p>StartScreenImage title</p>', () => {
    const result = wrapper.vm.htmlSerializer('paragraph', null, null, ['StartScreenImage title'])
    expect(result).toBe('<p>StartScreenImage title</p>')
  })

  it('if type preformatted will return <pre>StartScreenImage title</pre>', () => {
    const result = wrapper.vm.htmlSerializer('preformatted', null, null, ['StartScreenImage title'])
    expect(result).toBe('<pre>StartScreenImage title</pre>')
  })

  it('if type strong will return <strong>StartScreenImage title</strong>', () => {
    const result = wrapper.vm.htmlSerializer('strong', null, null, ['StartScreenImage title'])
    expect(result).toBe('<strong>StartScreenImage title</strong>')
  })

  it('if type em will return <em>StartScreenImage title</em>', () => {
    const result = wrapper.vm.htmlSerializer('em', null, null, ['StartScreenImage title'])
    expect(result).toBe('<em>StartScreenImage title</em>')
  })

  it('if type list-item will return <li>StartScreenImage title</li>', () => {
    const result = wrapper.vm.htmlSerializer('list-item', null, null, ['StartScreenImage title'])
    expect(result).toBe('<li>StartScreenImage title</li>')
  })

  it('if type o-list-item will return <li>StartScreenImage title</li>', () => {
    const result = wrapper.vm.htmlSerializer('o-list-item', null, null, ['StartScreenImage title'])
    expect(result).toBe('<li>StartScreenImage title</li>')
  })

  it('if type group-list-item will return <ul>StartScreenImage title</ul>', () => {
    const result = wrapper.vm.htmlSerializer('group-list-item', null, null, ['StartScreenImage title'])
    expect(result).toBe('<ul>StartScreenImage title</ul>')
  })

  it('if type group-o-list-item will return <ol>StartScreenImage title</ol>', () => {
    const result = wrapper.vm.htmlSerializer('group-o-list-item', null, null, ['StartScreenImage title'])
    expect(result).toBe('<ol>StartScreenImage title</ol>')
  })

  it('if type span and content null will return empty string', () => {
    const result = wrapper.vm.htmlSerializer('span', null, null, ['StartScreenImage title'])
    expect(result).toBe('')
  })

  it('if type span will return "StartScreenImage <br /> title"', () => {
    const result = wrapper.vm.htmlSerializer('span', null, 'StartScreenImage \n title', ['StartScreenImage title'])
    expect(result).toBe('StartScreenImage <br /> title')
  })

  it('if type embed will return embed html', () => {
    const element = {
      oembed: {
        provider_name: '',
        type: '',
        embed_url: '',
        html: 'html',
      },
    }
    const embedHtml = '<div data-oembed="" data-oembed-type="" data-oembed-provider="" ></div>'
    const result = wrapper.vm.htmlSerializer('embed', element, null, ['StartScreenImage title'])
    expect(toOneLine(result)).toStrictEqual(toOneLine(embedHtml))
  })

  describe('hyperlink tests', () => {
    it('if type hyperlink & without target will return <a href=""></a>', () => {
      const element = {
        data: {
          target: null,
        },
      }
      const hyperlinkHtml = '<a href=""></a>'
      const result = wrapper.vm.htmlSerializer('hyperlink', element, null, ['StartScreenImage title'])
      expect(toOneLine(result)).toStrictEqual(toOneLine(hyperlinkHtml))
    })

    it('if type hyperlink will return <a target="_blank" rel="noopener" href=""></a>', () => {
      const element = {
        data: {
          target: '_blank',
        },
      }

      const hyperlinkHtml = '<a target="_blank" rel="noopener" href=""></a>'
      const result = wrapper.vm.htmlSerializer('hyperlink', element, null, ['StartScreenImage title'])
      expect(toOneLine(result)).toStrictEqual(toOneLine(hyperlinkHtml))
    })
  })

  describe('label tests', () => {
    it('if type label will return <span class="label"></span>', () => {
      const element = {
        data: {
          label: 'label',
        },
      }
      const labelHtml = '<span class="label"></span>'
      const result = wrapper.vm.htmlSerializer('label', element, null, ['StartScreenImage title'])
      expect(toOneLine(result)).toStrictEqual(toOneLine(labelHtml))
    })

    it('if type label & without label will return <span ></span>', () => {
      const element = {
        data: {
          label: null,
        },
      }
      const labelHtml = '<span ></span>'
      const result = wrapper.vm.htmlSerializer('label', element, null, ['StartScreenImage title'])
      expect(toOneLine(result)).toStrictEqual(toOneLine(labelHtml))
    })
  })

  it('if type image will return html for img', () => {
    const element = {
      linkTo: {
        target: '_blank',
      },
      label: '',
      url: '',
      alt: '',
    }
    const imageHtml = '<p class=" block-img"><img src="" alt="Image" copyright=""></p>'
    const result = wrapper.vm.htmlSerializer('image', element, null, ['StartScreenImage title'])
    expect(toOneLine(result)).toStrictEqual(toOneLine(imageHtml))
  })
})
