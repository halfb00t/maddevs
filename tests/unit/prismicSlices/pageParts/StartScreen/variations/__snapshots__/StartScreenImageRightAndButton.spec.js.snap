// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreenImageRightAndButton component should correctly render startScreenImageRightAndButton 1`] = `
<div>
  <section
    class="start-screen-slice start-screen-slice-with-image"
    data-testid="start-screen-slice-with-image"
    id="transparent-header-area"
  >
    <img
      alt="img"
      class="start-screen-slice__image"
      data-testid="start-screen-slice__image"
      height="969"
      src="img.jpg"
      style="opacity: 0.9;"
      width="1680"
    />
     
    <div
      class="container"
    >
      <div
        class="start-screen-slice__flex-wrapper"
        style="opacity: 1;"
      >
        <div
          class="start-screen-slice__content"
        >
          <div
            class="start-screen-slice__titles-wrapper"
          >
            <prismicrichtext-stub
              class="start-screen-slice__title"
              data-testid="start-screen-slice__title"
              field="[object Object]"
              html-serializer="function () { [native code] }"
            />
          </div>
           
          <prismicrichtext-stub
            class="start-screen-slice__subtitle"
            data-testid="start-screen-slice__subtitle"
            field="[object Object]"
            html-serializer="function () { [native code] }"
          />
           
          <div
            class="start-screen-slice__btn-box"
          >
            <uibutton-stub
              class="start-screen-slice__btn"
              to="/"
            >
              
            Test text
            
              <img
                alt="Arrow"
                height="20"
                src="@/assets/img/common/arrow-up-right.svg"
                width="20"
              />
            </uibutton-stub>
          </div>
        </div>
         
        <div
          class="start-screen-slice__img-box"
        >
          <img
            alt="img"
            class="start-screen-slice__img-box-image"
            data-testid="right-image"
            height="380.25"
            src="img.jpg"
            width="259.69"
          />
        </div>
      </div>
    </div>
     
    <transition-stub
      class="modal-ebook"
      mode="out-in"
      name="main-fade"
    >
      <!---->
    </transition-stub>
  </section>
</div>
`;
